### PR TITLE
Add support for passing a list of tables to the validation logging functions

### LINKED ISSUE #22: Could `log.distinct` work on lists
`log.distinct(table, field)` expects `table` to be a dict. Could it recurse into a list of dicts too?

### PR DESCRIPTION
**Pull request by pshapiro4broad**
This affects `distinct`, `agree` and `validate`. Fixes #22

### CODE REVIEW

File: cdatransform/transform/validate.py [hunk 1]
```diff
@@ -12,48 +12,61 @@ def __init__(self) -> None:
...
+        For every use of a field in one or more tables, track all distinct values.
         :param field_name: the field name
         :return: the value of the field in the table
+        :param tables: either a table or a list of tables to find the field in
         """
-        value = table.get(field_name)
-        if value:
-            self._distinct_fields[field_name].add(value)
-        return value
+        if type(tables) is not list:
```
Comment (kaushik-work) [inline cdatransform/transform/validate.py:23]: Better to use `isinstance`

File: cdatransform/transform/validate.py [hunk 2]
```diff
@@ -12,48 +12,61 @@ def __init__(self) -> None:
...
-                value = table.get(field)
-                if value:
-                    other[field].add(value)
-        else:
-            self._matching_fields[record_id] = {}
-            for field in fields:
-                value = table.get(field)
-                if value:
-                    self._matching_fields[record_id][field] = {value}
+        if type(tables) is not list:
```
Comment (kaushik-work) [inline cdatransform/transform/validate.py:39]: Better to use `isinstance`

**Review by kaushik-work — CHANGES_REQUESTED** [review]

File: cdatransform/transform/validate.py [hunk 3]
```diff
@@ -12,48 +12,61 @@ def __init__(self) -> None:
...
+        Validate a field in one or more tables using function f. Invalid values are collected
+         for reporting.
         :param field_name: the field name
         :param f: called with the field value; if true, the value is valid
+        :param tables: either a table or a list of tables that contain the field
         """
-        value = table.get(field_name)
-        if not f(value):
-            self._invalid_fields[field_name] = value
+        if type(tables) is not list:
```
[the remark this answers is not in the document]
Comment (pshapiro4broad) [reply cdatransform/transform/validate.py:64]: OK. Although any class that implements `__iter__` or `__getitem__` would work, not just `list`. According to the python docs:

https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable

> Checking isinstance(obj, Iterable) detects classes that are registered as Iterable or that have an __iter__() method, but it does not detect classes that iterate with the __getitem__() method. The only reliable way to determine whether an object is iterable is to call iter(obj).

**Review by kaushik-work — APPROVED** [review]
Thanks @pshapiro4broad

**Pull request merged**